opm/simulators/utils/GridDataOutput_impl.hpp

```diff
@@ -0,0 +1,524 @@
+/*
+  Copyright 2023 Inria, Bretagne–Atlantique Research Center
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <opm/simulators/utils/GridDataOutput.hpp>
+
+#include <dune/grid/common/rangegenerators.hh>
+#include <dune/grid/io/file/vtk/common.hh>
+
+#include <opm/common/ErrorMacros.hpp>
+
+#include <cstddef>
+#include <ostream>
+
+namespace Opm::GridDataOutput {
+
+template <class GridView, unsigned int partitions>
+SimMeshDataAccessor<GridView,partitions>::
+ SimMeshDataAccessor(const GridView& gridView, Dune::PartitionSet<partitions> dunePartition)
+    : gridView_(gridView)
+    , dunePartition_(dunePartition)
+{
+    dimw_ = GridView::dimension; // this is an enum
+    partition_value_ = dunePartition.value;
+    countEntities();
+}
+
+template <class GridView, unsigned int partitions>
+bool SimMeshDataAccessor<GridView,partitions>::polyhedralCellPresent() const
+{
+    for (const auto& cit : elements(gridView_, dunePartition_)) {
+        auto corner_geom = cit.geometry();
+        if (Dune::VTK::geometryType(corner_geom.type()) == Dune::VTK::polyhedron) {
+            return true;
+        }
+    }
+    return false;
+}
+
+template <class GridView, unsigned int partitions>
+void SimMeshDataAccessor<GridView,partitions>::countEntities()
+{
+    // We include all the vertices for this ranks partition
+    const auto& vert_partition = vertices(gridView_, Dune::Partitions::all);
+    nvertices_ = std::distance(vert_partition.begin(), vert_partition.end());
```

Comment:
I think we need to `#include <algorithm>` to bring in a declaration of `std::distance()`.

Reply:
actually it's defined in ```<iterator>``` - included that one.